src/containers/HomePage.js

```diff
@@ -36,18 +38,37 @@ class HomePage extends React.Component {
     });
   }
 
+  selectTarget = (index) => {
+    if (!this.props.match.isExact) history.push(routes.index);
+    this.setState({
+      selectedTarget: index,
+    });
+  }
+
+  deleteSelectedTarget = () => {
+    const selectedTargetID = this.props.targets[this.state.selectedTarget].target.id;
+    this.props.deleteTarget({ id: selectedTargetID, index: this.state.selectedTarget });
+    this.setState({
+      selectedTarget: null
+    });
+  }
+
   mapClick = ({ latLng }) => {
     this.setState({
-      target: { lat: latLng.lat(), lng: latLng.lng() }
+      target: { lat: latLng.lat(), lng: latLng.lng() },
+      selectedTarget: null
     });
     if (this.props.match.isExact) history.push(routes.newTarget);
   }
 
   render() {
     const { location, subRoutes, targets } = this.props;
-    const { target, coords } = this.state;
+    const { target, coords, selectedTarget } = this.state;
     return (
       <div className="home-page">
+        {this.state.selectedTarget != undefined &&
```

Comment:
You are already taking `selectedTarget` from `this.state` destructuring above.

Also, `selectedTarget != undefined` would be the same to just `selectedTarget &&`

Reply:
selectedTarget is an array index. When the selection is the first item in the array it's 0 and evaluates to false as a boolean.

Reply:
Makes sense